apps/verticalhome/js/clock.js

```diff
@@ -36,27 +36,95 @@ function Clock() {
   this.timerID = null;
 
   /**
-   * Start the timer used to refresh the clock, will call the specified
-   * callback at every timer tick to refresh the UI. The callback used to
-   * refresh the UI will also be called immediately to ensure the UI is
-   * consistent.
+   * Contains time format (12 Hour or 24 Hour)
+   * @type {string from localization}
+   */
+  this.timeFormat = null;
+
+  /**
+   * Dom Object that contains time
+   * @type {Dom element}
+   */
+  this.clockTime = document.getElementById('clock-time');
+
+  /**
+   * This dom object contains date information
+   * @type {date}
+   */
+  this.date = document.getElementById('date');
+
+  window.addEventListener('timeformatchange', this);
+
+  /**
+   * Handle the event which fires when time format is chaned
+   * @param  {object} e [Event object]
+   */
+  this.handleEvent = function cl_handleEvents(e){
+
+    switch(e.type) {
+      case 'timeformatchange':
+        if (!this.l10nready) {
+          return;
+        }
+
+        this.timeFormat = window.navigator.mozHour12 ?
+          navigator.mozL10n.get('shortTimeFormat12') :
+          navigator.mozL10n.get('shortTimeFormat24');
+        this.refreshClock(new Date());
+
+        break;
+    };
+
+  };
+
+  /**
+   * We need to do some refreshing thing after l10n is ready.
+   *
+   * @memberof LockScreen
```

Comment:
and still LockScreen smell in your commit. BTW why for this particular function you specified '@memberof' tag, and for others no? Probably it'll be better to align documentation either delete here, or add for remaining functions.